drivers/include/at.h

```diff
@@ -53,55 +54,91 @@ extern "C" {
 /**
  * @brief End of line character to send after the AT command.
  */
-#ifndef AT_SEND_EOL
-#define AT_SEND_EOL "\r"
+#if IS_ACTIVE(CONFIG_AT_SEND_EOL_WINDOWS)
+#define CONFIG_AT_SEND_EOL   "\r\n"
+#elif IS_ACTIVE(CONFIG_AT_SEND_EOL_UNIX)
+#define CONFIG_AT_SEND_EOL   "\n"
+#elif IS_ACTIVE(CONFIG_AT_SEND_EOL_MAC)
+#define CONFIG_AT_SEND_EOL   "\r"
+#endif
+
+#ifndef CONFIG_AT_SEND_EOL
+#define CONFIG_AT_SEND_EOL "\r"
+#endif
+
+/**
+ * @brief Enable this to disable check for echo after an AT
+ * command is sent.
+ */
+#ifdef DOXYGEN
+#define CONFIG_AT_SEND_SKIP_ECHO
 #endif
 
 /**
  * @brief Enable/disable the expected echo after an AT command is sent.
+ *
+ * @deprecated Use inverse @ref CONFIG_AT_SEND_SKIP_ECHO instead.
+ * Will be removed after 2021.01 release.
  */
 #ifndef AT_SEND_ECHO
+#if IS_ACTIVE(CONFIG_AT_SEND_SKIP_ECHO)
+#define AT_SEND_ECHO 0
+#else
 #define AT_SEND_ECHO 1
 #endif
+#endif
 
 /**
- * @brief 1st end of line character received (S3 aka CR character for a modem).
+ * @brief default OK reply of an AT device.
  */
-#ifndef AT_RECV_EOL_1
-#define AT_RECV_EOL_1   "\r"
+#ifndef CONFIG_AT_RECV_OK
+#define CONFIG_AT_RECV_OK "OK"
 #endif
 
 /**
- * @brief 1st end of line character received (S4 aka LF character for a modem).
+ * @brief default ERROR reply of an AT device.
  */
-#ifndef AT_RECV_EOL_2
-#define AT_RECV_EOL_2   "\n"
+#ifndef CONFIG_AT_RECV_ERROR
+#define CONFIG_AT_RECV_ERROR "ERROR"
 #endif
 
+#if defined(MODULE_AT_URC) || DOXYGEN
+
 /**
- * @brief default OK reply of an AT device.
+ * @brief   Default buffer size used to process unsolicited result code data.
+ *          (as exponent of 2^n).
+ * 
+ *          As the buffer size ALWAYS needs to be power of two, this option
+ *          represents the exponent of 2^n, which will be used as the size of
+ *          the buffer.
  */
-#ifndef AT_RECV_OK
-#define AT_RECV_OK "OK"
+#ifndef CONFIG_AT_BUF_SIZE_EXP
+#define CONFIG_AT_BUF_SIZE_EXP (7)
 #endif
+/** @} */
 
 /**
- * @brief default ERROR reply of an AT device.
+ * @brief   Size of buffer used to process unsolicited result code data.
  */
-#ifndef AT_RECV_ERROR
-#define AT_RECV_ERROR "ERROR"
-#endif
-/** @} */
+#define AT_BUF_SIZE   (1<<CONFIG_AT_BUF_SIZE_EXP)
 
-/** Shortcut for getting send end of line length */
-#define AT_SEND_EOL_LEN  (sizeof(AT_SEND_EOL) - 1)
+/**
+ * @brief 1st end of line character received (S3 aka CR character for a modem).
+ */
+#ifndef AT_RECV_EOL_1
+#define AT_RECV_EOL_1   "\r"
```

Comment:
Why these two are not in Kconfig?

Reply:
Two reasons.
1. Kconfig does not allow escape sequences Refer Issue [here](https://github.com/ulfalizer/Kconfiglib/issues/91). Hence we will need to create bool symbols for several escape sequences.
2. These are expected to be standard values. AT_RECV_EOL_1 is S3( Command Line Termination Character ).  Most devices expect it as a CR (\r) - ASCII 13. Also AT_RECV_EOL_2 is S4 ( Response Formatting Character ). Most devices expect it as LF/NL (\n) - ASCII 10

Reply:
> 2. These are expected to be standard values

The range of this character is from 0 to 127, this is just the default one, but a device could have a different one.

>  Kconfig does not allow escape sequences

Ok. But as these values are still configurable via `CFLAGS` move them to the configuration documentation group please.

Reply:
Okay. Is it necessary to move it to config_ namespace since it's not configurable via Kconfig?

Reply:
I think just having it in the correct config group should do it for now.